include/rand.h

```diff
@@ -0,0 +1,23 @@
+#ifndef RAND_H
+#define RAND_H
+
+#include "PR/ultratypes.h"
+
+//! These values are recommended by the algorithms book *Numerical Recipes in C. The Art of Scientific Computing*, 2nd
+//! Edition, 1992, ISBN 0-521-43108-5. (p. 284):
+//! > This is about as good as any 32-bit linear congruential generator, entirely adequate for many uses.
+#define RAND_MULTIPLIER 1664525
+#define RAND_INCREMENT 1013904223
+
+u32 Rand_Next(void);
+void Rand_Seed(u32 seed);
+f32 Rand_ZeroOne(void);
+f32 Rand_Centered(void);
+void Rand_Seed_Variable(u32* rndNum, u32 seed);
+u32 Rand_Next_Variable(u32* rndNum);
+f32 Rand_ZeroOne_Variable(u32* rndNum);
+f32 Rand_Centered_Variable(u32* rndNum);
+
+extern u32 gRandFloat;
+
+#endif
```

Comment:
Out of curiosity does this header work for OoT as well?

Reply:
It can yeah, the main difference is that `RAND_MULTIPLIER`, `RAND_INCREMENT`, and `sRandFloat` in OoT are all localized to `rand.c`, so they don't need to be put in the header. MM decided to use these for this one specific function in kankyo, instead of just using the `Rand` function provided to them... for whatever reason.

That means I renamed `sRandFloat` to `gRandFloat` and it needs to be externed in MM while it doesn't need to be in OoT. Guess it's up to OoT if they want to prioritize syncing over local variables, but it might not be the exact same